crates/extensions/c8y_mapper_ext/src/inventory.rs

```diff
@@ -50,12 +41,14 @@ impl CumulocityConverter {
 
         if let JsonValue::Object(map) = inventory_base {
             for (key, value) in map {
-                self.update_inventory_model(key.clone(), value.clone());
-                let mapped_message = self.register_twin_data_message(
-                    &EntityTopicId::default_main_device(),
+                let main_device_tid = EntityTopicId::default_main_device();
```

Comment:
Instead of assuming default naming, it would be better to request the entity store.

```suggestion
                let main_device_tid = self.entity_store.main_device;
```